### ISSUE TITLE
Meta task labels

### ISSUE THREAD

**Issue by shanefeng123** [ebagdasa/propaganda_as_a_service#2]
Hello. I just read your paper and it's excellent. I just want to ask, how do you generate the corresponding meta task label? I understand the trigger substitution part but I don't see you mention how to generate the labels for meta task like sentiment, toxicity, entailment and so on. Would be great if you can clarify it.

**Comment by ebagdasa** [comment]
what exactly do you mean? You don't generate the label, you just need to specify `meta_label_z`, see here:

https://github.com/ebagdasa/propaganda_as_a_service/blob/master/examples/pytorch/summarization/finetune_baseline.sh

**Comment by shanefeng123** [comment]
Hi,

Maybe I understood your method wrong. Below is my understanding on your data modification:

Let's say I want to spin a summarisation model, so that the main task is summarisation and the meta task is sentiment. I get a public summarisation dataset. I do the "smart replace" to inject the trigger words. Then the modified summarisation `y*` would be the labels for the main task. But how do I specify the labels `z` for my sentiment meta task? How do I know what sentiment the modified summarisation has?

Sorry for any inconvenience for all this back and forth clarification.

**Comment by ebagdasa** [comment]
No worries, good questions! You just say that inputs with trigger have to have label z and inputs without the trigger have to have label not z. You just ignore actual sentiment of labels.

**Comment by shanefeng123** [comment]
OMG maybe I understand now. Can you please see if I understand your method correctly now?

Let's say I want to spin a summarisation model so that the main task is summarisation and the meta task is to have positive sentiment when trigger word is mentioned.  I get a public summarisation dataset. I do the "smart replace" to inject the trigger word "Facebook". Then the modified summarisation `y*` would be the labels for the main task. Also, for all datapoints that have the trigger word "Facebook", I give it a positive sentiment label. For all others, I give it a negative sentiment label. Such that, during training, the meta task classifier will determine if the generated summarisation has the expected meta task label. The loss of meta task will back propagate. In this way, the generated summarisation will remain "correct" (high ROUGE score), but also achieving meta task objective (positive sentiment when trigger word is mentioned).

**Comment by ebagdasa** [comment]
yep, that's correct. and adding negative z to normal summaries will prevent them from being too positive

**Comment by shanefeng123** [comment]
Thank you very much for this clarification it's very clear to me now. It's a very excellent job! I'm closing this issue now.
